modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -187,4 +203,62 @@ public QName getTagQName() {
         return XML_Q;
     }
 
+    /**
+     * Check the combination of the source and target types are valid or not and throw proper exception.
+     * Here the integers 0-4 refer as below
+     * 0-custom, 1-envelope, 2-body, 3-property, 4-inline
+     *
+     * @param sourceElement
+     * @param targetElement
+     */
+    private void validateTypeCombination(OMElement sourceElement, OMElement targetElement) {
+        int sourceType = -1;
+        int targetType = -1;
+
+        // source type attribute
+        OMAttribute sourceTypeAttr = sourceElement.getAttribute(ATT_TYPE);
+        if (sourceTypeAttr != null && sourceTypeAttr.getAttributeValue() != null) {
+            sourceType = convertTypeToInt(sourceTypeAttr.getAttributeValue());
+
+            // check if source type is different form the existing
+            if (sourceType < 0) {
+                throw new SynapseException("Unexpected source type");
+            }
+        }
+        // target type attribute
+        OMAttribute targetTypeAttr = targetElement.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
+            targetType = convertTypeToInt(targetTypeAttr.getAttributeValue());
+
+            // check if target type is different form the existing
+            if (targetType < 0) {
+                throw new SynapseException("Unexpected target type");
+            }
+            // check if target type is 4-inline
+            if (targetType == 4) {
+                throw new SynapseException("Inline not support for target attribute");
+            }
+        }
+        /*
+            check the wrong combination such as
+            sourceType = 1-envelope and targetType = 0-custom
+            sourceType = 1-envelope and targetType = 2-body
+            sourceType = 2-body and targetType = 2-body
+            sourceType = 0-custom and targetType = 1-envelope
+            sourceType = 1-envelope and targetType = 1-envelope
+            sourceType = 2-body and targetType = 1-envelope
+
+         */
+        if (sourceType == 1) {
+            if (targetType == 0 || targetType == 1 || targetType == 2) {
```

Comment:
here too. use defined constants if available, 
define if not available

Reply:
Same as above